test/test_microstructures.jl

```diff
@@ -0,0 +1,37 @@
+@testset "microstructures" begin
+    ref_shape = RefHexahedron
+    order = 1
+
+    ip_fsn = Lagrange{ref_shape, order}()^3
+    ip_geo = Lagrange{ref_shape, order}()^3
+    cv = CellValues(QuadratureRule{ref_shape}(1), ip_fsn, ip_geo)
+
+    ring_grid = generate_ring_mesh(8,2,2)
```

Comment:
The issue which you observed was that the grid is too coarse to capture the microstructure with sufficient accuracy.
```suggestion
    ring_grid = generate_ring_mesh(20,2,2)
```